optuna/trial/_fixed.py

```diff
@@ -108,16 +108,23 @@ def suggest_discrete_uniform(self, name: str, low: float, high: float, q: float)
         discrete = distributions.DiscreteUniformDistribution(low=low, high=high, q=q)
         return self._suggest(name, discrete)
 
-    def suggest_int(self, name, low, high, step=1, log=False):
-        # type: (str, int, int, int, bool) -> int
-        if log:
-            sample = self._suggest(
-                name, distributions.IntLogUniformDistribution(low=low, high=high, step=step)
-            )
+    def suggest_int(self, name: str, low: int, high: int, step: int = 1, log: bool = False) -> int:
+        if step != 1:
+            if log:
+                raise ValueError("The parameter `step >= 2` is not supported when `log` is True.")
```

Comment:
Actually, why don't you do this check in `IntLogUniformDistribution.__init__` to keep input validation consistent in terms of location and to not have to implement the same logic in various trails? I guess removing the argument altogether from `IntLogUniformDistribution.__init__` and `IntLogUniformDistribution.step` is a topic that should be addressed too to not confuse users of this distribution such as  sampler implementations but that may be outside the scope of this PR.

Reply:
Thank you for your insightful comment. I totally agree with your idea to remove the `step` argument from `IntLogUniformDistribution`, but it might be out of this PR. I will create another PR after this PR.